File: e-goi-wrapper-php/Egoi/XmlRpcImpl.php
<?php
if (!defined("EgoiApiXmlRpcImpl")) {
	define("EgoiApiXmlRpcImpl",0);

	require_once 'Zend/XmlRpc/Client.php';
	
	define('XmlRpcUrl', 'http://api.e-goi.com/v2/xmlrpc.php');
	
	class EgoiApiXmlRpcImpl extends EgoiApi {

		var $rpc;
		
		function __construct() {
			$this->rpc = new Zend_XmlRpc_Client(XmlRpcUrl);
		}
		
		function addExtraField($map) {
			return $this->rpc->call(__FUNCTION__, array($map));
		}
		
		function addSubscriber($map) {
			return $this->rpc->call(__FUNCTION__, array($map));
		}
		
		
		function addSubscriberBulk($map) {
			return $this->rpc->call(__FUNCTION__, array($map));
		}
		
		
		function checklogin($map) {
			return $this->rpc->call(__FUNCTION__, array($map));
		}
		
		
		function createCampaignEmail($map) {
			return $this->rpc->call(__FUNCTION__, array($map));
		}
		
		
		function createCampaignFAX($map) {
			return $this->rpc->call(__FUNCTION__, array($map));
		}
		
		
		function createCampaignSMS($map) {
			return $this->rpc->call(__FUNCTION__, array($map));
		}
		
		
		function createCampaignVoice($map) {
			return $this->rpc->call(__FUNCTION__, array($map));
		}
		
		
		function createList($map) {
			return $this->rpc->call(__FUNCTION__, array($map));
		}
		
		
		function createSegment($map) {
			return $this->rpc->call(__FUNCTION__, array($map));
		}
		
		
		function deleteCampaign($map) {
			return $this->rpc->call(__FUNCTION__, array($map));
		}
		
		
		function deleteSegment($map) {
			return $this->rpc->call(__FUNCTION__, array($map));
		}
		
		
		function editCampaignEmail($map) {
			return $this->rpc->call(__FUNCTION__, array($map));
		}
		
		
		function editCampaignSMS($map) {
			return $this->rpc->call(__FUNCTION__, array($map));
		}
		
		
		function editExtraField($map) {
			return $this->rpc->call(__FUNCTION__, array($map));
		}
		
		
		function editSubscriber($map) {
			return $this->rpc->call(__FUNCTION__, array($map));
		}
		
		
		function getCampaigns($map) {
			return $this->rpc->call(__FUNCTION__, array($map));
		}
		
		
		function getClientData($map) {
			return $this->rpc->call(__FUNCTION__, array($map));
		}
		
		
		function getCredits($map) {
			return $this->rpc->call(__FUNCTION__, array($map));
		}
		
		
		function getLists($map) {
			return $this->rpc->call(__FUNCTION__, array($map));
		}
		
		
		function getReport($map) {
			return $this->rpc->call(__FUNCTION__, array($map));
		}
		
		
		function getSegments($map) {
			return $this->rpc->call(__FUNCTION__, array($map));
		}
		
		
		function getSenders($map) {
			return $this->rpc->call(__FUNCTION__, array($map));
		}
		
		
		function getUserData($map) {
			return $this->rpc->call(__FUNCTION__, array($map));
		}
		
		
		function removeSubscriber($map) {
			return $this->rpc->call(__FUNCTION__, array($map));
		}
		
		
		function sendCall($map) {
			return $this->rpc->call(__FUNCTION__, array($map));
		}
		
		
		function sendEmail($map) {
			return $this->rpc->call(__FUNCTION__, array($map));
		}
		
		
		function sendFAX($map) {
			return $this->rpc->call(__FUNCTION__, array($map));
		}
		
		
		function sendSMS($map) {
			return $this->rpc->call(__FUNCTION__, array($map));
		}
		
		
		function subscriberData($map) {
			return $this->rpc->call(__FUNCTION__, array($map));
		}
		
		
		function updateList($map) {
			return $this->rpc->call(__FUNCTION__, array($map));
		}
	}
	
}
?>

File: e-goi-wrapper-python/EgoiApi/XmlRpcImpl.py
# -*- coding: utf-8 -*-
from xmlrpclib import ServerProxy

class EgoiApiXmlRpc:
    
    def __init__(self, url):
        self.client = ServerProxy(url) 
    
    def getLists(self, functionOptions):
        return self.client.getLists(functionOptions)

    def addExtraField(self, functionOptions):
        return self.client.addExtraField(functionOptions)

    def addSubscriber(self, functionOptions):
        return self.client.addSubscriber(functionOptions)

    def addSubscriberBulk(self, functionOptions):
        return self.client.addSubscriberBulk(functionOptions)

    def checklogin(self, functionOptions):
        return self.client.checklogin(functionOptions)

    def createCampaignEmail(self, functionOptions):
        return self.client.createCampaignEmail(functionOptions)

    def createCampaignFAX(self, functionOptions):
        return self.client.createCampaignFAX(functionOptions)

    def createCampaignSMS(self, functionOptions):
        return self.client.createCampaignSMS(functionOptions)

    def createCampaignVoice(self, functionOptions):
        return self.client.createCampaignVoice(functionOptions)

    def createList(self, functionOptions):
        return self.client.createList(functionOptions)

    def createSegment(self, functionOptions):
        return self.client.createSegment(functionOptions)

    def deleteCampaign(self, functionOptions):
        return self.client.deleteCampaign(functionOptions)

    def deleteSegment(self, functionOptions):
        return self.client.deleteSegment(functionOptions)

    def editCampaignEmail(self, functionOptions):
        return self.client.editCampaignEmail(functionOptions)

    def editCampaignSMS(self, functionOptions):
        return self.client.editCampaignSMS(functionOptions)

    def editExtraField(self, functionOptions):
        return self.client.editExtraField(functionOptions)

    def editSubscriber(self, functionOptions):
        return self.client.editSubscriber(functionOptions)

    def getCampaigns(self, functionOptions):
        return self.client.getCampaigns(functionOptions)

    def getClientData(self, functionOptions):
        return self.client.getClientData(functionOptions)

    def getCredits(self, functionOptions):
        return self.client.getCredits(functionOptions)

    def getReport(self, functionOptions):
        return self.client.getReport(functionOptions)

    def getSegments(self, functionOptions):
        return self.client.getSegments(functionOptions)

    def getSenders(self, functionOptions):
        return self.client.getSenders(functionOptions)

    def getUserData(self, functionOptions):
        return self.client.getUserData(functionOptions)

    def removeSubscriber(self, functionOptions):
        return self.client.removeSubscriber(functionOptions)

    def sendCall(self, functionOptions):
        return self.client.sendCall(functionOptions)

    def sendEmail(self, functionOptions):
        return self.client.sendEmail(functionOptions)

    def sendFAX(self, functionOptions):
        return self.client.sendFAX(functionOptions)

    def sendSMS(self, functionOptions):
        return self.client.sendSMS(functionOptions)

    def subscriberData(self, functionOptions):
        return self.client.subscriberData(functionOptions)

    def updateList(self, functionOptions):
        return self.client.updateList(functionOptions)

